### ISSUE TITLE
Empty relationships should have  data key with value null

### ISSUE THREAD

**Issue by JakeDluhy** [danivek/json-api-serializer#38]
Per spec: http://jsonapi.org/format/#document-resource-object-relationships `A “relationship object” MUST contain at least one of the following: links, data, meta`. Implementations I've seen in the past will have `"posts": { "data": null } }` within the relationships object if the relation doesn't exist/isn't provided.

However with schema like
```javascript
module.exports = {
  whitelist:     ['name'],
  relationships: {
    players: { type: 'players' },
    turns:   { type: 'turns' },
    map:     { type: 'maps' },
  },
};
```

my  json comes out like
```js
{
    "jsonapi": {
        "version": "1.0"
    },
    "data": {
        "type": "campaigns",
        "id": "10",
        "attributes": {
            "name": "My New Campaign"
        },
        "relationships": {
            "players": {},
            "turns": {},
            "map": {}
        }
    }
}
```

I'm saying it should be like
```js
"relationships": {
    "players": { "data": null },
    "turns": { "data": null },
    "map": { "data": null }
}
```

I looked through the source to see if I could make a PR but I couldn't find where this is happening... It seems like at the very least from https://github.com/danivek/json-api-serializer/blob/master/lib/JSONAPISerializer.js#L582 it should have `data: undefined`

**Comment by danivek** [comment]
You're right.
It's related to this commit: https://github.com/danivek/json-api-serializer/commit/12bc2cd4e4b0cbd1628b874a41b26d7ca9eda6e8

Return null instead of undefined will do the job i think.

**Comment by JakeDluhy** [comment]
Want me to make a PR for it?

**Comment by danivek** [comment]
fix in v1.10.1

Thanks

**Comment by wesselvdv** [comment]
Hi guys, this change is really giving me headaches. I know you're following the spec with this, but it's causing the following situation:

```typescript
const JSONAPISerializer = require('json-api-serializer');
const Serializer = new JSONAPISerializer();

// Schools type
Serializer.register('schools', {
  relationships: {
    students: {
      type: 'students',
      links: {
        self: '/students'
      }
    }
  }
});

// Students type
Serializer.register('students', {
  relationships: {
    grades: {
      type: 'grades',
      links: {
        self: '/grades'
      }
    }
  }
});

Serializer.register('grades');

// Input data for schools
const input = {
  "id": "1",
  "name": "School Name",
  "students": {
    id: 1,
  },
};

const output = Serializer.serialize('schools', input);
console.log(JSON.stringify(output));
```
```json
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": {
    "type": "schools",
    "id": "1",
    "attributes": {
      "name": "School Name"
    },
    "relationships": {
      "students": {
        "links": {
          "self": "/students"
        },
        "data": {
          "type": "students",
          "id": "1"
        }
      }
    }
  },
  "included": [
    {
      "type": "students",
      "id": "1",
      "attributes": {},
      "relationships": {
        "grades": {
          "links": {
            "self": "/grades"
          },
          "data": null
        }
      }
    }
  ]
}
```

See that this change requires me to always give back an object completely, instead of partially (For example when I am only updating a student its association to a school). The student might have grades, but they're just not relevant in this exchange. If a client receives this, and it previously received a complete students object with grades, it will REMOVE the relationship between the student and the grades because according to this partial exchange it no longer exists.

I hope it's understandable that I won't be happy if I have to started querying my database on an entity its relationships on every CRUD action to ensure integrity on the client side.
